### ISSUE TITLE
Question: What does "$d" on hover tell the user - and is it wanted behavior?

### ISSUE THREAD

**Issue by andys8** [haskell/haskell-language-server#3463]
Hey there,

what does `$d` in `$dMonadRandom :: MonadRandom m` stand for? Is it wanted and provides a meaning I'm not aware of - or is it some sort of leftover or broken regex?

Thanks for the help.

![image](https://user-images.githubusercontent.com/13085980/214055931-8c666231-1034-418b-8e37-4d5b66afaf6e.png)
![image](https://user-images.githubusercontent.com/13085980/214055967-57a829b8-d620-43a9-8997-d60beff745db.png)

## Your environment 

Vim + [CoC](https://github.com/andys8/dotfiles/blob/master/coc-settings.json)

```shell
haskell-language-server version: 1.9.0.0 (GHC: 9.2.5) (PATH: /home/andreas/.ghcup/bin/haskell-language-server-wrapper-1.9.0.0)
Tool versions found on the $PATH
cabal:          3.6.2.0
stack:          2.9.1
ghc:            9.2.5
```

```shell
vim --version
VIM - Vi IMproved 9.0 (2022 Jun 28, compiled Dec 11 2022 13:50:40)
Included patches: 1-1046
Compiled by Arch Linux
```

**Comment by fendor** [comment]
Names generated by GHC, #3316 aims to filter it out afaict

**Comment by andys8** [comment]
Cool, that's the hint I was hoping for. I tested the example with the build of the PR. It will filter out both lines. If that's what we want, then I consider this issue can be closed immediately or when #3316 is merged. Thanks.

![image](https://user-images.githubusercontent.com/13085980/214107661-dd08b94c-632b-4eb2-a4e6-c7613bf01e1d.png)

**Comment by fendor** [comment]
Clsoed by #3316
